File: Music.Model/Track.cs
using System.Collections.Generic;

namespace Music.Model
{
	public class Track
	{
		public int Id { get; set; }
		public int Number { get; set; }
		public string Name { get; set; }

		public string FileName { get; set; }

		public int DiscId { get; set; }
		public Disc Disc { get; set; }

		public int GenreId { get; set; }
		public Genre Genre { get; set; }

		public List<Contribution> Contributions { get; set; }
	}
}

File: Music.Web/ClientApp/src/app/models/track.ts
import { BaseModel } from './base-model';
import { Contribution } from './contribution';
import { Disc } from './disc';
import { Genre } from './genre';

export interface Track extends BaseModel {
  number: number;
  name: string;
  fileName: string;
  genreId: number;
  genre?: Genre;
  discId: number;
  disc?: Disc;
  contributions: Contribution[];
}
